src/lib/coerce.js

```diff
@@ -33,8 +37,18 @@ exports.valObjectMeta = {
         otherOpts: ['dflt'],
         coerceFunction: function(v, propOut, dflt) {
             // TODO maybe `v: {type: 'float32', vals: [/* ... */]}` also
-            if(isArrayOrTypedArray(v)) propOut.set(v);
-            else if(dflt !== undefined) propOut.set(dflt);
+            if(isArrayOrTypedArray(v)) {
+                propOut.set(v);
+            } else if(isPrimitiveTypedArrayRepr(v)) {
```

Comment:
Ideally, the `{dtype: '', data: ''}` -> typed array conversion should happen during the `calc` step. More precisely somewhere here:

https://github.com/plotly/plotly.js/blob/24a0f91a6fcbe6dfb7fbb68feecb6257c0bb32ba/src/plots/polar/set_convert.js#L103-L136

Depending on how slow this conversion can be, moving it to the calc step will help ensure faster interactions (note that the `calc` is skipped on e.g. zoom and pan). 

This might be a fairly big job though, you'll have to replace all the `isArrayOrTypedArray` calls upstream of the calc step with something like `isArrayOrTypedArrayOrIsPrimitiveTypedArrayRepr` (or something less verbose :smile: ).

So I guess, we should first benchmark `primitiveTypedArrayReprToTypedArray` what kind of potential perf gain we could get.

Reply:
@etpinard  Maybe I'm misunderstanding something. As it is now, I thought these conversions would be happening in the `supplyDefaults` logic for the various traces. Does that logic run on events like zoom/pan?

Either way, I'll get some performance numbers for `primitiveTypedArrayReprToTypedArray`. I was hoping I wouldn't need to retrace all of the steps you went through to add the original `TypedArray` support!